logprep/processor/amides/processor.py

```diff
@@ -0,0 +1,212 @@
+"""
+Adaptive Misuse Detection System (AMiDeS)
+-----------------------------------------
+
+The AMiDeS-processor performs Adaptive Misuse Detection on command-lines
+of incoming process creation events. In case of a positive result, additional
+rule attribution is performed.
+
+Example
+^^^^^^^
+
+..  code-block:: yaml
+    :linenos:
+
+    - amides:
+        type: amides
+        specific_rules:
+            - tests/testdata/rules/specific/
+        generic_rules:
+            - tests/testdata/rules/generic/
+        max_cache_entries: 10000
+        decision_threshold: 0.0
+        num_rule_attributions: 10
+        models_path: /tmp/models.zip
+"""
+from time import time
+from multiprocessing import current_process
+from typing import List, Tuple
+from logging import Logger
+from zipfile import ZipFile
+from pathlib import Path
+from functools import lru_cache, _lru_cache_wrapper
+import joblib
+
+from attr import define, field, validators
+
+from logprep.abc.processor import Processor
+from logprep.metrics.metric import calculate_new_average
+from logprep.processor.amides.rule import AmidesRule
+from logprep.processor.amides.detection import MisuseDetector, RuleAttributor
+from logprep.processor.amides.normalize import CommandLineNormalizer
+from logprep.util.helper import get_dotted_field_value
+from logprep.util.getter import GetterFactory
+
+
+class AmidesError(BaseException):
+    """Default exception for all Amides-related errors."""
+
+    def __init__(self, name: str, message: str):
+        super().__init__(f"Amides ({name}): {message}")
+
+
+class Amides(Processor):
+    """Detects if incoming command-lines are malicious or not. Performs additional
+    rule attribution in case of a malicious command-line."""
+
+    @define(kw_only=True)
+    class Config(Processor.Config):
+        """Amides processor configuration class."""
+
+        max_cache_entries: int = field(default=1048576, validator=validators.instance_of(int))
+        """Maximum number of cached command-lines and their classification results."""
+        decision_threshold: float = field(validator=validators.instance_of(float))
+        """Specifies the decision threshold of the misuse detector to adjust it's overall
+        classification performance."""
+        num_rule_attributions: int = field(default=10, validator=validators.instance_of(int))
+        """Number of rule attributions returned in case of a positive misuse detection result."""
+        models_path: str = field(validator=validators.instance_of(str))
+        """Path or URI of the archive (.zip) containing the models used by the misuse detector
+        and the rule attributor."""
+
+    @define(kw_only=True)
+    class AmidesMetrics(Processor.ProcessorMetrics):
```

Comment:
I wonder if you call it `ProcessorMetrics` and inherit from `Processor.ProcessorMetrics` there is a good chance, you do not have to initialize the the metrics here, because it is done in `Processor.__init__` already.